ufm-stream-to-fluentd/ufm-stream-to-fluentd.py

```diff
@@ -0,0 +1,218 @@
+    except Exception as e:
+        logging.error(e)
+
+
+def load_memory_with_jsons():
+    global stored_versioning_api
+    global stored_systems_api
+    global stored_ports_api
+    global stored_links_api
+
+    try:
+        logging.info(f'Call load_memory_with_jsons')
+
+        if os.path.exists(UFM_API_VERSIONING_RESULT):
+            stored_versioning_api = read_json_from_file(UFM_API_VERSIONING_RESULT)
+
+        if os.path.exists(UFM_API_SYSTEMS_RESULT):
+            stored_systems_api = read_json_from_file(UFM_API_SYSTEMS_RESULT)
+
+        if os.path.exists(UFM_API_PORTS_RESULT):
+            stored_ports_api = read_json_from_file(UFM_API_PORTS_RESULT)
+
+        if os.path.exists(UFM_API_LINKS_RESULT):
+            stored_links_api = read_json_from_file(UFM_API_LINKS_RESULT)
+    except Exception as e:
+        logging.error(e)
+
+
+def update_ufm_apis():
+    global stored_versioning_api
+    global stored_systems_api
+    global stored_ports_api
+    global stored_links_api
+
+    try:
+        logging.info(f'Call update_ufm_apis')
+
+        ufm_new_version = send_ufm_request(UFM_API_VERSIONING)
+
+        # check if systems api is changed
+        if (stored_versioning_api == '' or ufm_new_version["switches_version"] != stored_versioning_api[
+            "switches_version"] or
+                ufm_new_version["switches_version"] != stored_versioning_api["switches_version"]):
+            stored_systems_api = send_ufm_request(UFM_API_SYSTEMS)
+            write_json_to_file(UFM_API_SYSTEMS_RESULT, stored_systems_api)
+
+        # check if ports api is changed
+        if stored_versioning_api == '' or ufm_new_version["ports_version"] != stored_versioning_api["ports_version"]:
+            stored_ports_api = send_ufm_request(UFM_API_PORTS)
+            write_json_to_file(UFM_API_PORTS_RESULT, stored_ports_api)
+
+        # check if links api is changed
+        if stored_versioning_api == '' or ufm_new_version["links_version"] != stored_versioning_api["links_version"]:
+            stored_links_api = send_ufm_request(UFM_API_LINKS)
+            write_json_to_file(UFM_API_LINKS_RESULT, stored_links_api)
+
+        stored_versioning_api = ufm_new_version
+    except Exception as e:
+        logging.error(e)
+
+
+def send_ufm_request(url):
+    url = ufm_protocol + '://' + ufm_host + '/ufmRest/' + url
+    headers = {}
+    # token auth: to be done
+    # if token:
+    #    headers = {"Authorization": "Bearer " + token['access_token']}
+    try:
+        logging.info(f'Send UFM API Request, URL: {url}')
+        response = requests.get(url, verify=False, headers=headers, auth=(ufm_username, ufm_password))
+        logging.info("UFM API Request Status [" + str(response.status_code) + "], URL " + url)
+        if response.raise_for_status():
+            logging.error(response.raise_for_status())
+    except Exception as e:
+        logging.error(e)
+    return response.json()
```

Comment:
If the request has failed, trying to convert the response to JSON may throw an exception.
Return different value (e.g. empty dict) in this case or handle the exception in higher levels.